src/rideables/Stack.hpp

```diff
@@ -0,0 +1,71 @@
+// #ifndef STACK_HPP
+// #define STACK_HPP
+
+#include <stdio.h>
+#include<iostream>
+#include <atomic>
+
+using namespace std;
+
+class Stack
+{
+    private:
+
+    typedef struct StackNode
+    {
+
+        int data;
+        StackNode *next;
+
+    } * StackNodePtr;
+
+    atomic<StackNodePtr> top;
+
+    public:
+    Stack()
+    {
+        top = NULL;
+    }
+    void push(int data){
+
```

Comment:
When you read code in Michael's textbook, you may notice that there's another field `counter` in `top`. This is to prevent ABA problem (You can read the textbook in page 124 to know what it is). 
You can create a template class called `CntPtr<T>`, defined as:
```c++
template <class T>
class CntPtr{
    T* ptr;
    uint64_t cnt;
};
```
Every time you do `pop`, the counter in `top` should increment.

Another way to prevent this kind of ABA problem is to use some safe memory reclamation (SMR) mechanism: to ensure that a node is reclaimed and repurposed only after operations potentially using it finishes, this ABA problem will no longer happen. This may be hard to understand but don't worry, we can talk about this later.
If you take a look at other nonblocking data structures implemented in this repo (e.g., [MS queue](https://github.com/urcs-sync/Montage/blob/master/src/rideables/MSQueue.hpp)), you will see a `RCUTracker` used at the beginning and the end of each operation, and call `retire` rather than C++ `delete` when we want to reclaim nodes no longer in use. Do similar thing here with no anti-ABA counter would also work.

The first paragraph in the textbook page 25 talks a little bit about SMR.